tests/integration/bond_test.py

```diff
@@ -553,3 +556,57 @@ def test_set_miimon_100_on_existing_bond(bond99_with_2_slaves):
     bond_config[Bond.OPTIONS_SUBTREE] = {"miimon": 100}
     libnmstate.apply(state)
     assertlib.assert_state_match(state)
+
+
+@pytest.fixture
+def eth1_eth2_with_no_profile():
+    iface_down(ETH1)
+    iface_down(ETH2)
+    yield
+    iface_down(ETH1)
+    iface_down(ETH2)
+
+
+def _nmcli_profile_reactivate(ifname):
+    """
+    Use nmcli to deactivate the profile and reactivate
+    """
+    assert cmdlib.exec_cmd(["nmcli", "connection", "down", ifname])[0] == 0
+    # Wait slave been deactivated
+    time.sleep(1)
+    assert cmdlib.exec_cmd(["nmcli", "connection", "up", ifname])[0] == 0
+    # Wait slave been activated
+    time.sleep(1)
+
+
+def test_bond_mac_be_consistent_with_diff_slave_order(
+    eth1_eth2_with_no_profile,
+):
+    """
+    When sever reboot, network manager will by default activate slaves in the
+    order of their name. Nmstate should provide the consistent MAC address for
+    bond regardless the order of slaves.
+    Use nmcli to reactivate the profile to simulate the server reboot.
+    """
+    eth1_mac = get_mac_address(ETH1)
+    with bond_interface(
+        name=BOND99,
+        slaves=[ETH2, ETH1],
```

Comment:
Isn't this similar to `test_reordering_the_slaves_does_not_change_the_mac`?

Reply:
The old one `test_reordering_the_slaves_does_not_change_the_mac` does not actually show the failure as the eth1/eth2 already has profile created at the time of bond creation.

Old test case removed.

Reply:
If the old test shows the expected behavior for existing bonds, it should be kept but probably be renamed to properly describe what it is checking. IMHO it would also be better if there was a comment with a user story that describes the expected behavior since the test names do not seem to properly capture this. AFAIU, It would be

When I change the order of slaves for an existing bond in the desired state, the MAC address does not change.